pkg/flowaggregator/flowaggregator.go

```diff
@@ -213,40 +231,138 @@ type flowAggregator struct {
 	sendJSONRecord              bool
 	numRecordsExported          int64
 	numRecordsReceived          int64
+	updateCh                    chan updateMsg
+	configFile                  string
+	configWatcher               *fsnotify.Watcher
+	configData                  []byte
+	APIServer                   flowaggregatorconfig.APIServerConfig
 }
 
 func NewFlowAggregator(
-	externalFlowCollectorAddr string,
-	externalFlowCollectorProto string,
-	activeFlowRecTimeout time.Duration,
-	inactiveFlowRecTimeout time.Duration,
-	aggregatorTransportProtocol AggregatorTransportProtocol,
-	flowAggregatorAddress string,
-	includePodLabels bool,
 	k8sClient kubernetes.Interface,
-	observationDomainID uint32,
 	podInformer coreinformers.PodInformer,
-	sendJSONRecord bool,
-) *flowAggregator {
+	configFile string,
+) (*flowAggregator, error) {
+	if len(configFile) == 0 {
+		return nil, fmt.Errorf("configFile is empty string")
+	}
 	registry := ipfix.NewIPFIXRegistry()
 	registry.LoadRegistry()
+
+	var err error
+	configWatcher, err := fsnotify.NewWatcher()
+	if err != nil {
+		return nil, fmt.Errorf("error when creating file watcher for configuration file: %v", err)
+	}
+	if err = configWatcher.Add(filepath.Dir(configFile)); err != nil {
```

Comment:
I recommend adding a comment here explaining why we are watching the directory and not the configuration file (or rather symlink) directly

Reply:
Added